lib/plug/crypto.ex

```diff
@@ -132,15 +128,24 @@ defmodule Plug.Crypto do
   """
   @spec secure_compare(binary(), binary()) :: boolean()
   def secure_compare(left, right) when is_binary(left) and is_binary(right) do
-    byte_size(left) == byte_size(right) and secure_compare(left, right, 0)
+    byte_size(left) == byte_size(right) and crypto_hash_equals(left, right)
+  end
+
+  defp crypto_hash_equals(x, y) do
+    # TODO: remove when we require OTP 25.0
+    if Code.ensure_loaded?(:crypto) and function_exported?(:crypto, :hash_equals, 2) do
```

Comment:
I think we can move this to compile-time!

Reply:
Moved